protostar/starknet/cheatable_execute_entry_point.py

```diff
@@ -300,6 +305,26 @@ def append_runtime_profile(
             ContractProfile(current_callstack, entry_point, profile)
         )
 
+    async def execute_for_testing(
+        self,
+        state: State,
+        general_config: StarknetGeneralConfig,
+        resources_manager: Optional[ExecutionResourcesManager] = None,
+        tx_execution_context: Optional[TransactionExecutionContext] = None,
+    ) -> CallInfo:
+
+        new_config = deepcopy(general_config)
+        if self.max_steps is not None:
+
+            # Providing a negative value to Protostar results in infinite steps,
+            # this is here to mimic default Cairo behavior
+            value = None if self.max_steps < 0 else self.max_steps
+            new_config.__dict__["invoke_tx_max_n_steps"] = value
+
+        return await super().execute_for_testing(
+            state, new_config, resources_manager, tx_execution_context
+        )
```

Comment:
```suggestion
        # Providing a negative value to Protostar results in infinite steps,
        # this is here to mimic default Cairo behavior.
        if self.max_steps is not None and self.max_steps >= 0:
            general_config = dataclasses.replace(
                general_config,
                invoke_tx_max_n_steps=value
            )

        return await super().execute_for_testing(
            state, general_config, resources_manager, tx_execution_context
        )
```

I think this should work?

Reply:
It will result in a default value (aka not infinity) for negative numbers

Reply:
You can tweak the condition here, but mostly I wanted to express usage of `dataclasses.replace` here

Reply:
`dataclasses.replace` seems to be blocked by the typeguard, it allows setting this value to int only, and sometimes None is needed

Reply:
`# type: ignore`?

Reply:
`typeguard` raises `TypeError` either way

Reply:
OK, please leave a `# NOTE:` why you did this code that way though.